pkgs/development/ruby-modules/gem-config/default.nix

```diff
@@ -87,6 +87,14 @@ in
       sed -i $installPath/lib/dep-selector-libgecode.rb -e 's@VENDORED_GECODE_DIR =.*@VENDORED_GECODE_DIR = "${gecode_3}"@'
     '';
   };
+  
+  ethon = attrs: {
+    dontBuild = false;
+    postPatch = ''
+      substituteInPlace lib/ethon/curls/settings.rb \
+        --replace "libcurl" "${curl.out}/lib/libcurl.so"
```

Comment:
do we care about Darwin?

Reply:
`stdenv.hostPlatform.extensions.sharedLibrary` would contains the correct extension in this case.